File: src/main/java/net/lexicodes/kesandra/events/chat/commands/MainPluginCommand.java
package net.lexicodes.kesandra.events.chat.commands;

import net.lexicodes.kesandra.helpers.MenuUtils;
import org.bukkit.ChatColor;
import org.bukkit.command.Command;
import org.bukkit.command.CommandExecutor;
import org.bukkit.command.CommandSender;
import org.bukkit.entity.Player;
import net.lexicodes.kesandra.Kesandra;
import net.lexicodes.kesandra.helpers.Utilities;

public class MainPluginCommand implements CommandExecutor{

    @Override
    public boolean onCommand(CommandSender sender, Command command, String label, String[] args) {

        // verify that the user has proper permissions
        if (!sender.hasPermission("template.user")) {
            Utilities.warnPlayer(sender, Kesandra.getPhrase("no-permissions-message"));
            return true;
        }

        try {

            switch (args[0].toLowerCase()) {
                case "help":
                    help(sender);
                    break;
                case "info":
                    info(sender);
                    break;
                case "tutorial":
                    if (sender instanceof Player) MenuUtils.tutorialMenu((Player) sender);
                    else Utilities.warnPlayer(sender, Kesandra.getPhrase("no-console-message"));
                    break;

                // put plugin specific commands here

                case "reload":
                    if (sender.hasPermission("template.admin")) reload(sender);
                    else Utilities.warnPlayer(sender, Kesandra.getPhrase("no-permissions-message"));
                    break;
                default:
                    Utilities.warnPlayer(sender, Kesandra.getPhrase("not-a-command-message"));
                    help(sender);
                    break;
            }

        } catch(Exception e) {
            Utilities.warnPlayer(sender, Kesandra.getPhrase("formatting-error-message"));
        }

        return true;
    }

    private void info(CommandSender sender) {
        sender.sendMessage(Kesandra.prefix + ChatColor.GRAY + "Plugin Info");
        sender.sendMessage(ChatColor.DARK_PURPLE + "- " + ChatColor.GREEN + "Version " + Kesandra.getInstance().getVersion() + " - By ThirtyVirus");
        sender.sendMessage("");
        sender.sendMessage(ChatColor.DARK_PURPLE + "- " + ChatColor.GREEN + "~The best plugin template ever!");
        sender.sendMessage("");
        sender.sendMessage(ChatColor.DARK_PURPLE + "- " + ChatColor.RESET + ChatColor.RED + "" + ChatColor.BOLD + "You" + ChatColor.WHITE + ChatColor.BOLD + "Tube" + ChatColor.GREEN + " - https://youtube.com/thirtyvirus");
        sender.sendMessage(ChatColor.DARK_PURPLE + "- " + ChatColor.RESET + ChatColor.DARK_AQUA + "" + ChatColor.BOLD + "Twitter" + ChatColor.GREEN + " - https://twitter.com/thirtyvirus");
        sender.sendMessage(ChatColor.DARK_PURPLE + "- " + ChatColor.RESET + ChatColor.GOLD + "" + ChatColor.BOLD + "SpigotMC" + ChatColor.GREEN + " - https://spigotmc.org/members/179587/");
        sender.sendMessage(ChatColor.DARK_PURPLE + "------------------------------");
    }

    private void help(CommandSender sender) {
        sender.sendMessage(Kesandra.prefix + ChatColor.GRAY + "Commands");
        sender.sendMessage(ChatColor.DARK_PURPLE + "- " + ChatColor.GRAY + "/template help");
        sender.sendMessage(ChatColor.DARK_PURPLE + "- " + ChatColor.GRAY + "/template info");
        sender.sendMessage(ChatColor.DARK_PURPLE + "- " + ChatColor.GRAY + "/template tutorial");
        sender.sendMessage(ChatColor.DARK_PURPLE + "- " + ChatColor.GRAY + "/template reload");
        sender.sendMessage(ChatColor.DARK_PURPLE + "------------------------------");
    }

    private void reload(CommandSender sender) {
        Kesandra.getInstance().reloadConfig();
        Kesandra.getInstance().loadConfiguration();
        Kesandra.getInstance().loadLangFile();

        Utilities.informPlayer(sender, "configuration, values, and language settings reloaded");
    }

}

File: src/main/java/net/lexicodes/kesandra/events/chat/commands/MainPluginCommand.kt
package net.lexicodes.kesandra.events.chat.commands

import net.lexicodes.kesandra.Kesandra
import net.lexicodes.kesandra.helpers.Utilities
import org.bukkit.ChatColor
import org.bukkit.command.Command
import org.bukkit.command.CommandExecutor
import org.bukkit.command.CommandSender
import java.util.*

class MainPluginCommand : CommandExecutor {
    override fun onCommand(sender: CommandSender, command: Command, label: String, args: Array<String>): Boolean {

        // verify that the user has proper permissions
        if (!sender.hasPermission("template.user")) {
            Utilities.warnPlayer(sender, Kesandra.Companion.getPhrase("no-permissions-message"))
            return true
        }
        try {
            when (args[0].lowercase(Locale.getDefault())) {
                "help" -> help(sender)
                "info" -> info(sender)
                "reload" -> if (sender.hasPermission("template.admin")) reload(sender) else Utilities.warnPlayer(
                    sender,
                    Kesandra.Companion.getPhrase("no-permissions-message")
                )

                else -> {
                    Utilities.warnPlayer(sender, Kesandra.Companion.getPhrase("not-a-command-message"))
                    help(sender)
                }
            }
        } catch (e: Exception) {
            Utilities.warnPlayer(sender, Kesandra.Companion.getPhrase("formatting-error-message"))
        }
        return true
    }

    private fun info(sender: CommandSender) {
        sender.sendMessage(ChatColor.DARK_PURPLE.toString() + "------------------------------")
        sender.sendMessage(Kesandra.Companion.prefix + ChatColor.GRAY + "Plugin Info")
        sender.sendMessage(
            ChatColor.DARK_PURPLE.toString() + "- " + ChatColor.GREEN + "Version " + Kesandra.Companion.getInstance()
                .getVersion() + " - By LexiCodes And Almondz"
        )
        sender.sendMessage(ChatColor.DARK_PURPLE.toString() + "------------------------------")
    }

    private fun tutorial(sender: CommandSender) {
        sender.sendMessage(ChatColor.DARK_PURPLE.toString() + "------------------------------")
        sender.sendMessage(Kesandra.Companion.prefix + ChatColor.GRAY + "Plugin Info")
        sender.sendMessage(
            ChatColor.DARK_PURPLE.toString() + "- " + ChatColor.GREEN + "Version " + Kesandra.Companion.getInstance()
                .getVersion() + " - By LexiCodes And Almondz"
        )
        sender.sendMessage(ChatColor.DARK_PURPLE.toString() + "------------------------------")
    }

    private fun help(sender: CommandSender) {
        sender.sendMessage(ChatColor.DARK_PURPLE.toString() + "------------------------------")
        sender.sendMessage(Kesandra.Companion.prefix + ChatColor.GRAY + "Commands")
        sender.sendMessage(ChatColor.DARK_PURPLE.toString() + "- " + ChatColor.GRAY + "/template help")
        sender.sendMessage(ChatColor.DARK_PURPLE.toString() + "- " + ChatColor.GRAY + "/template info")
        sender.sendMessage(ChatColor.DARK_PURPLE.toString() + "- " + ChatColor.GRAY + "/template tutorial")
        sender.sendMessage(ChatColor.DARK_PURPLE.toString() + "- " + ChatColor.GRAY + "/template reload")
        sender.sendMessage(ChatColor.DARK_PURPLE.toString() + "------------------------------")
    }

    private fun reload(sender: CommandSender) {
        Kesandra.Companion.getInstance().reloadConfig()
        Kesandra.Companion.getInstance().loadConfiguration()
        Kesandra.Companion.getInstance().loadLangFile()
        Utilities.informPlayer(sender, "configuration, values, and language settings reloaded")
    }
}